new/server-stats-endpoint.md

```diff
@@ -0,0 +1,115 @@
+for Chef Server and all its co-located services.
+
+
+## Motivation
+
+    As a Chef Server operator,
+    I want to have a consistent way to get important Chef Server metrics,
+    so that I can easily monitor the health of my Chef Server out of the box.
+
+    As a Chef Server operator,
+    I want to understand the metrics I'm given,
+    so that I can easily monitor the health of my Chef Server.
+
+
+## Specification
+
+This RFC adds a series of `/_stats/$service` endpoints to Chef Server. Each service
+colocated on the box will be provided an endpoint, and with respond to GET requests.
+The response format will be a text format which is specified by 
+[Prometheus 0.0.4](https://prometheus.io/docs/instrumenting/exposition_formats/#text-format-details).
+This gives us the ability to use a few metrics types specified below along with the ability to give
+a description for any metrics which are provided.
+
+### Supported Metric Primitives
+The endpoints can provide metrics of following types, as defined by [Prometheus](https://prometheus.io/docs/concepts/metric_types/):
+- Counter - Monotonic cumulative metric. For example, number of requests served.
+- Gauge - Numeric measurement at a given point in time. For example, number of inflight requests.
+- Histogram - Distributes values into buckets. For example, response times.
+- Summary - Provides summary statistics for a value in a sliding window.
+
+### Response Format
+#### Why Use the Prometheus Format?
+The format is already defined and provides metric types that are known to work with at least
+one monitoring system. This format also gives us the ability to describe each metric presented,
+so the end user would know what it means. The format itself is human readable and easy to parse,
+making it easy to both get started with and write tooling around.
+
+Using this format directly also could allow us to use metric exporters already written for other co-located
+services such as PostgreSQL, RabbitMQ, and Solr.
+
+#### Why Not?
+This would introduce an inconsistency in that the current API uses JSON and this one route would not. We
+could just as well create a JSON specification with information such as type, description, metric name, etc.
+
+
+### Services
+At a minimum, the Chef Server frontend service would need to provide stats. For example, `bookshelf`, `oc-id`,
+`oc_bifrost`, and `erchef` should provide metrics as those are always colocated. Since these run together, these could also be grouped into a single metrics endpoint(`_stats/chef-server`) to reduce the number of calls needed to gather the metrics.
+
+If Chef Server is configured to provide services such as RabbitMQ, Solr, Elasticsearch, and PostgreSQL, an
+endpoint should be provided for each service.
+
+#### Service List
+Since different implementations or configurations of Chef Server could have different dependencies, 
+`GET /_stats` will provide a list of services.
+
+Example response:
+
+```
+{
```

Comment:
This might be best, and would also make returning a JSON format easier if we decided we wanted to.